### DISCUSSION TITLE
Setting priority faces doesn't do anything.

### DISCUSSION THREAD

**Discussion by niceiq** [minad/org-modern#231] [Q&A]
Pretty simple issue here. The following code does not do anything at all: 

```
     org-modern-priority-faces
     '((?A . (:foreground "black" :background "#68aaba")) ; Red
       (?B . (:foreground "black" :background "#d48c0d")) ; Orange
       (?C . (:foreground "black" :background "#d6ba04")) ; Yellow
       (?D . (:foreground "black" :background "#93e9fa"))) ; Blue

```

Modifying the string works as expected: 

```
     org-modern-priority
     '((?A . "󱈸ASAP󱈸")
       (?B . "HIGH")
       (?C . "MEDIUM")
       (?D . "LOW"))
```

Solution I've found so far is to use: 

```
    (setq org-priority-faces '(
                             (?A . (:background "black" :foreground "#b85862" :weight 'bold))
                             (?B . (:background "black" :foreground "#d48c0d"))
                             (?C . (:background "black" :foreground "#d6ba04"))
                             (?D . (:background "black" :foreground "#93e9fa"))))
```

Is there a difference between those two variables? 

Also, is is possible to modify the font to be of similar size to the TODO item? 

![image](https://github.com/user-attachments/assets/953738f3-b909-4467-8dc9-eea5edd2ee1c)

**Comment by minad** [comment]
> The following code does not do anything at all:

Modifying the faces works for me. I think the problem is that you cannot configure both `org-modern-priority` and `org-modern-priority-faces` together. If you configure `org-modern-priority` it must already carry the face information.

**Comment by niceiq** [comment]
So is there a way to change both the colors and the string?

**Reply by minad** [reply]
You can use `(propertize "str" 'face ...)`.

**Comment by rallyemax** [comment]
@minad , could you please provide a working example? I tried various combinations, and whenever defining both the string and face in `org-modern-priority`, the resulting face defaulted to an unstyled version of whatever `org-priority` provided.

I'm sure the fault is on my end -- I've never really grokked the face properties system. Here is the last variant I tried with `org-modern-priority`:
```elisp
        org-modern-priority '((?0 .
                                (propertize "P0"
                                            'face '('(:foreground "DarkRed" :background "LightPink")
                                                    'org-modern-priority))))
```
My goal was to use your `org-modern-priority` face but to add those fore- and back-ground colors. (I had other levels defined as well, but keeping the example short.)

My current workaround is to just define the strings in `org-modern-priority`:
```elisp
    (setq org-modern-priority '((?0 . " P0 ")
                                (?1 . " P1 ") 
                                         ...)
```
> The one-char padding around the strings is necessary because by defining things this way,
> `org-modern-priority` doesn't add the padding characters.

And then to inherit from your `org-modern-priority` face in `org-priority-faces`:
```elisp
    (setq org-priority-faces '((?0 :inherit org-modern-priority 
                                   :background "DarkRed" :foreground "LightPink")
                               (?1 :inherit org-modern-priority 
                                   :background "DarkOrange4" :foreground "LightGoldenRod")
                                                                                      ...)
```
This works well enough -- the styling matches the `org-modern` TODO flag style. The only thing to look out for is that the `:foreground` and `:background` colors need to be reversed, I believe because the `org-modern` face is inverted.

Still, I'd like to achieve this result "properly" by just defining `org-modern-priority` if possible.

Love this project, by the way.

**Reply by jdtsmith** [reply]
That's a lot of extra `'` quotes and parens.  An alist of the sort you want could look like:

```elisp
`((?0 . ,(propertize "P0" 'face '(:foreground "DarkRed" :background "LightPink" org-modern-priority)))
```
